src/main/java/Deadline.java

```diff
@@ -0,0 +1,46 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoUnit;
+
+public class Deadline extends Task {
+    private String dueDate;
+    private String dueDateInSaveFormat;
+    private Deadline(String name, String dueDate, String dueDateInSaveFormat) {
+        super(name);
+        this.dueDate = dueDate;
+        this.dueDateInSaveFormat = dueDateInSaveFormat;
+    }
+
+    public static Deadline createDeadline(String name, String dueDate) throws DukeException{
+        LocalDate localDate;
+        try {
+            localDate = LocalDate.parse(dueDate);
+        }
+        catch (DateTimeParseException ex) {
+            throw new DukeException("Please specify the due date as follows: yyyy-mm-dd");
+        }
+        return new Deadline(name,
+                localDate.format(DateTimeFormatter.ofPattern("MMM d yyyy")), dueDate);
+    }
+
+    @Override
+    public String missingNameError() {
```

Comment:
Should this method be named as a verb? I've noticed this issue in several other places too.